x/dex/contract/execution.go

```diff
@@ -56,9 +56,17 @@ func ExecutePair(
 	typedContractAddr := dextypesutils.ContractAddress(contractAddr)
 	typedPairStr := dextypesutils.GetPairString(&pair)
 
+	// First cancel orders
 	cancelForPair(ctx, typedContractAddr, typedPairStr, dexkeeper, orderbook)
+	// Add all limit orders to the orderbook
+	orders := dexkeeper.MemState.GetBlockOrders(ctx, typedContractAddr, typedPairStr)
+	limitBuys := orders.GetLimitOrders(types.PositionDirection_LONG)
+	limitSells := orders.GetLimitOrders(types.PositionDirection_SHORT)
+	exchange.AddOutstandingLimitOrdersToOrderbook(orderbook, limitBuys, limitSells)
```

Comment:
nit: can you move line 62-64 to the `AddOutstandingLimitOrdersToOrderbook` function

Reply:
@LCyson this was done on purpose. It was more modular to pass in `limitBuys` and `limitSells` into `AddOutstandingLimitOrdersToOrderbook` rather than having `AddOutstandingLimitOrdersToOrderbook` take in the dex keeper and call that itself. More importantly, it makes it significantly easier to test `AddOutstandingLimitOrdersToOrderbook` and the limit order matching functionality.

Reply:
Resolved offline on discord